### PR TITLE
Fix(@rjsf/antd): Check for errors.length

### LINKED ISSUE #2575: @rjsf/antd Text input shows error icon when extra errors is `[]`
### Prerequisites

- [x] I have read the [documentation](https://react-jsonschema-form.readthedocs.io/);
- [x] In the case of a bug report, I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [x] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net/n1k0/f2y3fq7L/6/) or a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### Description
I ran into a scenario where there are no actual form errors, and the form can be submitted, but to the user it looks like the input is invalid. It has a red border and red error icon, indicating that it is invalid, even though it is not actually invalid.

### Steps to Reproduce
Here is sandbox: https://stackblitz.com/edit/react-1dcgru?file=src%2FApp.js

1. Use `@rjsf/antd`
2. Render a form component...

with an object schema with a string property. Example:
```js
{
  type: 'object',
    properties: {
      name: {
        type: 'string',
      }
    }
}
```
and a `extraErrors` prop with an empty array of errors. Example:
```js
{
  name: {
    __errors: [],
  },
}
```
3. Look at the red text input

#### Expected behavior
When the errors are `[]`, it should understand that there are no errors.

#### Actual behavior
The antd design input showed validation errors.
Validation error screenshot

### Version
```json
{
  "@ant-design/icons":"4.6.4",
  "@babel/runtime":"^7.15.4",
  "@rjsf/antd":"3.1.0",
  "@rjsf/core":"3.1.0",
  "antd":"4.16.13",
  "antd-dayjs-webpack-plugin":"1.0.0",
  "classnames":"^2.3.1",
  "dayjs":"1.10.7",
  "lodash":"4.17.21",
  "rc-picker":"^2.5.18",
  "react":"17.0.2",
  "react-dom":"17.0.2"
}
```

### PR DESCRIPTION
**Pull request by ChocolateLoverRaj**
### Reasons for making this change
Fixes #2575 

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [x] **I'm adding or updating code**
  - [x] I've added and/or updated tests
  - [x] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [x] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by ChocolateLoverRaj** [comment]
It's disappointing that after almost a year my bug fix didn't get merged and now the build is failing so I'm guessing it's even less likely to get merged.

**Comment by ChocolateLoverRaj** [comment]
Ok i will fix conflict

File: CHANGELOG.md [hunk 1]
```diff
@@ -272,6 +272,11 @@ should change the heading of the (upcoming) version to include a major version b
 - Added global `readonly` flag to the `Form` (https://github.com/rjsf-team/react-jsonschema-form/pull/2554)
 - Enable source maps in playground, for development (https://github.com/rjsf-team/react-jsonschema-form/pull/2568)
 
+# v3.1.1 (upcoming)
+
+## @rjsf/antd
+- Do not show errors if `extraErrors` has `[]` (https://github.com/rjsf-team/react-jsonschema-form/pull/2576).
+
```
Comment (heath-freenome) [inline CHANGELOG.md:279]: Can you move this up to the `5.0.0-beta.3` section?

File: packages/antd/test/EmptyErrorsArray.test.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import React from 'react';
```
Comment (heath-freenome) [inline packages/antd/test/EmptyErrorsArray.test.js:1]: Would you be willing to add this to the `Array.test.js` instead, at the end?

Comment (ChocolateLoverRaj) [reply packages/antd/test/EmptyErrorsArray.test.js:1]:
```js
import React from 'react'
```
is already in `Array.test.js`

Comment (heath-freenome) [reply packages/antd/test/EmptyErrorsArray.test.js:1]: I'm not sure you understand what I mean... You created a new `EmptyErrorsArray.test.js` file... Can you instead, copy the two tests and paste them at around line 77 in the `Array.test.js` file?

**Comment by heath-freenome** [comment]
Also, we are strictly checking formatting now, so you'll need to do `npm run cs-format` before you commit... Or you can setup husky to do this automatically on commit by doing `npm run prepare` at the root directory after doing `npm install`

**Comment by ChocolateLoverRaj** [comment]
@heath-freenome `npm i` and `pnpm i` both don't work. When `lerna bootstrap` does work, the tests fail because they can't import the dependencies. I can't work on this GH repo because of that.

**Comment by heath-freenome** [comment]
> @heath-freenome `npm i` and `pnpm i` both don't work. When `lerna bootstrap` does work, the tests fail because they can't import the dependencies. I can't work on this GH repo because of that.

That is odd... What error(s) are you seeing?

**Comment by ChocolateLoverRaj** [comment]
This is the error I get:
```
$ pnpm i
 WARN  deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated
Packages: +4 -910
+------------------------------------------------------------------------------------------------------------------------------------------------------------  
Progress: resolved 2697, reused 1145, downloaded 0, added 0, done

> react-jsonschema-form@5.0.0 postinstall C:\Users\rajas\Documents\GitHub\react-jsonschema-form
> lerna bootstrap

lerna notice cli v5.4.3
lerna info Bootstrapping 11 packages
lerna info Installing external dependencies
lerna info Symlinking packages and binaries
lerna WARN EREPLACE_EXIST @rjsf/core is already installed for @rjsf/semantic-ui. Replacing with symlink...
lerna WARN EREPLACE_EXIST @rjsf/utils is already installed for @rjsf/semantic-ui. Replacing with symlink...
lerna success Bootstrapped 11 packages

> react-jsonschema-form@5.0.0 prepare C:\Users\rajas\Documents\GitHub\react-jsonschema-form
> husky install

husky - Git hooks installed

dependencies:
- @ant-design/icons 4.7.0
- @babel/cli 7.18.10
- @babel/core 7.18.13
- @babel/plugin-proposal-class-properties 7.18.6  
- @babel/plugin-proposal-object-rest-spread 7.18.9
- @babel/plugin-proposal-optional-chaining 7.18.9 
- @babel/plugin-syntax-dynamic-import 7.8.3       
- @babel/plugin-transform-modules-commonjs 7.18.6 
- @babel/plugin-transform-object-assign 7.18.6    
- @babel/plugin-transform-react-jsx 7.18.10       
- @babel/plugin-transform-runtime 7.18.10
- @babel/preset-env 7.18.10
- @babel/preset-react 7.18.6
- @babel/register 7.18.9
- @babel/runtime 7.18.9
- @chakra-ui/icons 1.1.7
- @chakra-ui/react 1.8.8
- @emotion/cache 11.10.3
- @emotion/jest 11.10.0
- @emotion/react 11.10.0
- @emotion/styled 11.10.0
- @fluentui/react 7.196.0
- @material-ui/core 4.12.4
- @material-ui/icons 4.11.3
- @mui/icons-material 5.10.3
- @mui/material 5.10.3
- @rjsf/material-ui 5.0.0-beta.2
- @rollup/plugin-replace 4.0.0
- @types/jest-expect-message 1.0.4
- @types/json-schema 7.0.11
- @types/json-schema-merge-allof 0.6.1
- @types/lodash 4.14.184
- @types/react 17.0.48
- @types/react-dom 17.0.17
- @types/react-is 17.0.3
- @types/react-test-renderer 17.0.2
- ajv 6.12.6
- antd 4.22.8
- atob 2.1.2
- babel-loader 8.2.5
- chai 3.5.0
- chakra-react-select 3.3.8
- core-js 3.25.0
- cross-env 2.0.1
- css-loader 6.7.1
- dayjs 1.11.5
- estraverse 4.3.0
- estraverse-fb 1.3.2
- express 4.18.1
- framer-motion 5.6.0
- gh-pages 3.2.3
- html 1.0.0
- html-webpack-plugin 5.5.0
- jest 25.5.4
- jest-expect-message 1.0.2
- jest-watch-typeahead 2.1.1
- jsdom 16.7.0
- json-schema-merge-allof 0.8.1
- jsonpointer 5.0.1
- jss 10.9.2
- less 4.1.3
- less-loader 11.0.0
- loader-utils 3.2.0
- lodash 4.17.21
- lodash-es 4.17.21
- mini-css-extract-plugin 2.6.1
- mocha 5.2.0
- monaco-editor 0.33.0
- monaco-editor-webpack-plugin 7.0.1
- nanoid 3.3.4
- nyc 15.1.0
- prop-types 15.8.1
- react 17.0.2
- react-app-polyfill 1.0.6
- react-bootstrap 1.6.6
- react-dom 17.0.2
- react-frame-component 4.1.3
- react-icons 4.4.0
- react-is 18.2.0
- react-monaco-editor 0.49.0
- react-portal 4.2.2
- react-select 5.4.0
- react-test-renderer 17.0.2
- react-transform-catch-errors 1.0.2
- react-transform-hmr 1.0.4
- rimraf 3.0.2
- sass 1.54.6
- sass-loader 13.0.2
- semantic-ui-css 2.4.1
- semantic-ui-react 2.1.3
- sinon 9.2.4
- source-map-loader 4.0.0
- style-loader 3.3.1
- tsdx 0.14.1
- webpack 5.74.0
- webpack-cli 4.10.0
- webpack-dev-middleware 5.3.3
- webpack-dev-server 4.10.1
- webpack-hot-middleware 2.25.2

 ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies

.
├─┬ @babel/eslint-parser
│ └── ✕ missing peer @babel/core@>=7.11.0
├─┬ ts-jest
│ ├── ✕ missing peer @babel/core@">=7.0.0-beta.0 <8"
│ └── ✕ missing peer jest@^27.0.0
└─┬ dts-cli
  ├─┬ eslint-plugin-flowtype
  │ ├── ✕ missing peer @babel/plugin-syntax-flow@^7.14.5
  │ └── ✕ missing peer @babel/plugin-transform-react-jsx@^7.14.9
  ├─┬ rollup-plugin-sourcemaps
  │ └── ✕ missing peer @types/node@>=10.0.0
  └─┬ ts-node
    └── ✕ missing peer @types/node@"*"
Peer dependencies that should be installed:
  @babel/core@">=7.11.0 <8.0.0"              @babel/plugin-transform-react-jsx@^7.14.9  jest@^27.0.0
  @babel/plugin-syntax-flow@^7.14.5          @types/node@>=10.0.0

hint: If you want peer dependencies to be automatically installed, add "auto-install-peers=true" to an .npmrc file at the root of your project.
hint: If you don't want pnpm to fail on peer dependency issues, add "strict-peer-dependencies=false" to an .npmrc file at the root of your project.
```

**Comment by heath-freenome** [comment]
Have you tried using the `npm` version installed with node 16? That is working for me

**Comment by ChocolateLoverRaj** [comment]
I updated npm to latest version: `8.18.0`, and ran `npm i` (which took a loooong time) and there were no errors. This is the result of running test on `antd` dir:
```
 FAIL  test/Object.test.js
  ● Test suite failed to run

    Cannot find module '@rjsf/validator-ajv6' from 'test/Object.test.js'

      1 | import React from "react";
      2 | import renderer from "react-test-renderer";
    > 3 | import validator from "@rjsf/validator-ajv6";
        | ^
      4 |
      5 | import "../__mocks__/matchMedia.mock";
      6 | import Form from "../src";

      at Resolver.resolveModule (node_modules/dts-cli/node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (test/Object.test.js:3:1)
```

**Comment by heath-freenome** [comment]
Does this still happen after you do `npm run build`?

**Comment by ChocolateLoverRaj** [comment]
> Does this still happen after you do `npm run build`?
After running `npm run build` everything works. Thanks.

File: packages/antd/test/Array.test.js [hunk 3]
```diff
@@ -76,3 +76,60 @@ describe("array fields", () => {
     expect(tree).toMatchSnapshot();
   });
 });
+
+describe("errors", () => {
```
Comment (heath-freenome) [inline packages/antd/test/Array.test.js:80]: If you delete these three lines, then they will all be in a single describe block, which is what I was hoping for

**Review by heath-freenome — APPROVED** [review]

**Comment by heath-freenome** [comment]
@ChocolateLoverRaj All set, if you'd prefer I merge, then I can

**Comment by ChocolateLoverRaj** [comment]
Yes please merge

**Comment by heath-freenome** [comment]
> Yes please merge

Done, you'll have to delete the branch off of your fork yourself

**Pull request merged**